### ISSUE TITLE
GJS programs need to be wrapped specially

### ISSUE THREAD

**Issue by linsui** [NixOS/nixpkgs#318830]
### Describe the bug
See https://gitlab.gnome.org/GNOME/gjs/-/blob/1.80.2/modules/script/package.js#L38. The name of the js file is used to decide the resource file path. When the js file is wrapped by the wrapGAppsHook, the js file is moved to `.<original name>-wrapped` and a wrapper is generated to run this js file. The js file is run with shebang and the wrapped name is used by gjs. So the resource file can't be found.

### Steps To Reproduce
See https://github.com/NixOS/nixpkgs/pull/318829 for example.

### Expected behavior
When wrap the gjs script the file name is not changed.

### Notify maintainers

@NixOS/gnome 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by eclairevoyant** [comment]
the wrapper hook would use `makeBinaryWrapper`. as a workaround for now, have you tried with `dontWrapGApps = true;` + `wrapProgram` instead?

**Comment by linsui** [comment]
I thought makeBinaryWrapper is used by default and I don't know how it's supposed to make any difference. :shrug: I just tested and it doesn't work.

**Comment by eclairevoyant** [comment]
`make(Binary)Wrapper` takes 2 positional args, so that you can specify the wrapper name.

**Comment by jtojnar** [comment]
You need to patch `package._findEffectiveEntryPointName`, see for example:

https://github.com/NixOS/nixpkgs/blob/aa05442b89788a13a4620f8c71c27e20d449abe8/pkgs/desktops/gnome/core/gnome-shell/wrap-services.patch#L10-L12

**Comment by linsui** [comment]
Yep, another workaround... Can we have a wrapGjsHook which patch the file automatically?

**Comment by jtojnar** [comment]
It is difficult to automatically find what file to patch, and where to patch it.
